src/index.js

```
@@ -14,6 +15,25 @@ module.exports = class WebappWebpackPlugin {
       favicons: {},
       prefix: 'assets/',
     }, options);
+
+    this.tags = {};
+    this.tags.promise = new Promise((resolve, reject) => {
+      this.tags.resolve = resolve;
+      this.tags.reject = reject;
+    });
+    this.tags.promise.catch(() => {});
```

Are you sure we can place the `catch` here?

I believe it would swallow the rejection from the plugin and the loader `then` would execute with `undefined`. So we need the `catch` to come after the rule is created, to let the loader catch the rejection and report the error.

Maybe we need a test case for that.

---

If i did it like this:
```js
this.tags.promise = new Promise((resolve, reject) => {
  this.tags.resolve = resolve;
  this.tags.reject = reject;
}).catch(() => {});
```
Then yea, rejection will be shallowed and rule loader will get `undefined` in `then`.

Here though, because original promise is exposed (and not `catch`ed), both `catch`es are called on a rejection. (Checked it before.)

---

Here is the snippet to check in dev tools:
```js
class Plugin {
  constructor() {
    this.tags = {};
    this.tags.promise = new Promise((resolve, reject) => {
      this.tags.resolve = resolve;
      this.tags.reject = reject;
    });
    this.tags.promise.catch(() => {
      console.log('original catch');
    });
  }
}

const plugin = new Plugin();

plugin.tags.promise
  .then(() => console.log('new then'))
  .catch(() => console.log('new catch'));

plugin.tags.reject('error');
```

---

Oh I see, I didn't know one could _fork_ promisses into multiple continuations.